src/test/java/com/nerodesk/om/aws/AwsDocsTest.java

```
@@ -49,4 +61,57 @@ public void conformsToEqualsHashCodeContract() {
             .suppress(Warning.TRANSIENT_FIELDS)
             .verify();
     }
+
+    /**
+     * AwsDocs can list all docs.
+     * @throws IOException If something goes wrong.
+     */
+    @Test
+    public void listsDocs() throws IOException {
+        final Bucket bucket = this.mockBucket("lists", "lists-exists");
+        final List<String> expected = Lists.newArrayList(bucket.list(""));
+        final List<String> names = new AwsDocs(bucket).names();
+        MatcherAssert.assertThat(names, Matchers.equalTo(expected));
+        MatcherAssert.assertThat(names, Matchers.hasItem("sub/test"));
+    }
+
+    /**
+     * AwsDocs can find a doc.
+     * @throws IOException If something goes wrong.
+     */
+    @Test
+    public void findsDoc() throws IOException {
+        final String exists = "finds-exists";
+        final Bucket bucket = this.mockBucket("finds", exists);
+        final Docs docs = new AwsDocs(bucket);
+        MatcherAssert.assertThat(
+            docs.doc(exists).exists(),
+            Matchers.equalTo(true)
+        );
+        MatcherAssert.assertThat(
+            docs.doc("xyz").exists(),
+            Matchers.equalTo(false)
+        );
+    }
+
+    /**
+     * Builds a mock Bucket.
+     * @param name Bucket name.
+     * @param exists Name of Ocket that exists.
+     * @return The mock bucket.
+     * @throws IOException If something goes wrong.
+     */
+    private Bucket mockBucket(final String name, final String exists)
+        throws IOException {
+        final TemporaryFolder folder = new TemporaryFolder();
+        folder.create();
+        final File bucket = new File(folder.getRoot(), name);
+        assert bucket.mkdir();
```

we don't use asserts in our project. Please, use hamcrest if you need to check something.
In this particular case don't check at all. Fail fast